### PR TITLE
Fix DualShockUDP not adding/removing devices correctly

### PR DESCRIPTION
**Pull request by Filoppi**
-If adding 2 devices with the same name, their unique identifier wouldn't be increased, causing a conflict.
-Removing a device wouldn't actually remove it from the internal devices list because the local list of devices had already been updated when iterating it.
-It was possible to remove devices belonging to other input sources by adding a device with the same name and then removing it.

### CODE REVIEW

File: Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
 #include "InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPProto.h"
 
+#define DUALSHOCKUDP_SOURCE_NAME "DSUClient"
```
Comment (lioncash) [inline Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp:29]:
```suggestion
constexpr std::string_view DUALSHOCKUDP_SOURCE_NAME = "DSUClient"
```

This should also be moved down into the namespace.

Comment (Filoppi) [reply Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp:29]: True, though I copied the way from DInput.h, which declared its name outside of the namespace and as a define.

**Comment by iwubcode** [comment]
Huh, didn't ever imagine someone would use a duplicate device name.  That's certainly not intended and probably should be enforced.  But these changes look good.

**Comment by Filoppi** [comment]
> Huh, didn't ever imagine someone would use a duplicate device name. That's certainly not intended and probably should be enforced. But these changes look good.

I thought about enforcing it but that's not necessary given that two devices can have the exact same name under the same source, they will be distinguished by ID.

**Comment by iwubcode** [comment]
> I thought about enforcing it but that's not necessary given that two devices can have the exact same name under the same source, they will be distinguished by ID.

Yeah, not a problem from a software standpoint, more from a user standpoint.  Imagine you have two devices named the same thing and come back later.  How do you distinguish them?  I guess you have to look at the order.

More importantly, if I'm remembering right, profiles are based on device name.  I wonder what would happen if it was applied.  Maybe both would get it?

But like I said, it was never intended in my design, just an oversight :(

**Comment by Filoppi** [comment]
> > I thought about enforcing it but that's not necessary given that two devices can have the exact same name under the same source, they will be distinguished by ID.
> 
> Yeah, not a problem from a software standpoint, more from a user standpoint. Imagine you have two devices named the same thing and come back later. How do you distinguish them? I guess you have to look at the order.
> 
> More importantly, if I'm remembering right, profiles are based on device name. I wonder what would happen if it was applied. Maybe both would get it?
> 
> But like I said, it was never intended in my design, just an oversight :(

Well of course the first thing I tried to do was adding two devices with the same name to break it :).
Profiles seem to save the whole device path (e.g. DInput/0/Wireless Controller), when restoring it, if the device is not present, it will just add it back to the list and mark it as disconnected, all you got to do now is select a new default device (of the same type) and it would work, so not really a problem.
If I want to have two devices with the same name, it's my problem really, especially because if you attach two Xbox controllers, they will also have the same name, but a different ID.

File: Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp (hunk 1 above)
Comment (Filoppi) [reply Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp]: fixed

**Comment by iwubcode** [comment]
@Filoppi - sorry, I had just woken up when I looked at this initially and was thinking of the path not the actual device name.  You're right, there's no reason we can't allow the same name.

This LGTM

**Review by iwubcode — APPROVED** [review]

**Comment by AdmiralCurtiss** [comment]
Commits still need to be squashed.

**Comment by Filoppi** [comment]
> Commits still need to be squashed.

Fixed

**Comment by Filoppi** [comment]
Fixed lint code check

**Review by jordan-woyak — APPROVED** [review]

**Pull request merged**